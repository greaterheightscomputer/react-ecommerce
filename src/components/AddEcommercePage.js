import React from 'react';

const AddEcommercePage = () => (
    <div>
        This is from my add expense component
    </div>
);

export default AddEcommercePage;