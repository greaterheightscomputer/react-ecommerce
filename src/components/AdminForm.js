import React from 'react';
import { SingleDatePicker } from 'react-dates';
import moment from 'moment';

export default class AdminForm extends React.Component{    
    constructor(props) { //its will be use for both AddProduct and EditProduct
        super(props);

        this.state = {
            description: props.ecommerce ? props.ecommerce.description : '',
            category: props.ecommerce ? props.ecommerce.category : '',
            item: props.ecommerce ? props.ecommerce.item : '',
            amount: props.ecommerce ? (props.ecommerce.amount / 100).toString() : '',
            stock: props.ecommerce ? props.ecommerce.stock :'',       
            image: props.ecommerce ? props.ecommerce.image.name : '',
            imageUrl: props.ecommerce ? props.ecommerce.imageUrl : '',
            company: props.ecommerce ? props.ecommerce.company : '',
            info: props.ecommerce ? props.ecommerce.info: '',
            createdAt: props.ecommerce ? moment(props.ecommerce.createdAt) : moment(),
            focused: false,
            error: ''               
        }
    };
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    };
    onCategoryChange = (e) => {
        const category = e.target.value;
        this.setState(() => ({ category }));
    };
    onItemChange = (e) => {
        const item = e.target.value;
        this.setState(() => ({ item }));
    };
    onStockChange = (e) => {
        const stock = e.target.value;        
        this.setState(() => ({ stock }));
    };
    onAmountChange = (e) => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }));        
        }        
    };
    onCompanyChange = (e) => {
        const company = e.target.value;
        this.setState(() => ({ company }));
    };
    onInfoChange = (e) => {
        const info = e.target.value;
        this.setState(() => ({ info }));
    };
    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState(() => ({ createdAt }));            
        }        
    };
    onFocusChange = ({ focused}) => {
        this.setState(() => ({ focused }));
    };
    onImageChange = (e) => {        
        let reader = new FileReader();
        let file = e.target.files[0];        
        reader.onloadend = () => {
            this.setState({
                image: file,
                imageUrl: reader.result
            });
        };
        reader.readAsDataURL(file); 
    };
    onSubmitClick = (e) => {
        e.preventDefault();
        if (!this.state.description || !this.state.category || !this.state.item || !this.state.amount || !this.state.image || !this.state.stock || !this.state.company || !this.state.info) {        
            this.setState(() => ({
                error: 'Please provide value for the empty field(s)'
            }));    
        } else {
            this.setState(() => ({ error: '' }));            
            this.props.onSubmit({
                description: this.state.description,
                category: this.state.category,
                item: this.state.item,
                amount: parseFloat(this.state.amount, 10) * 100,
                stock: this.state.stock,
                image: this.state.image.name,
                imageUrl: this.state.imageUrl,
                company: this.state.company,
                info: this.state.info,
                createdAt: this.state.createdAt.valueOf()
            });            
        } 
    };
    render() {
        return (
            <form className="form" onSubmit={this.onSubmitClick}>                
                    <input
                        className= "text-input"
                        type="text"
                        placeholder="Description"
                        autoFocus
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                    />
                    <select
                        className= "select"
                        value={this.state.category}
                        onChange={this.onCategoryChange}                        
                    >
                        <option value="">Select Category</option>
                        <option value="boy">Boy</option>
                        <option value="girl">Girl</option>
                        <option value="amen">Men</option>
                        <option value="women">Women</option>
                    </select>      
                    <select
                        className= "select"
                        value={this.state.item}
                        onChange={this.onItemChange}>
                        <option value="">Select Item</option>
                        {/*boyItems */}
                        <option value="boy_bag">Boy Bag</option>
                        <option value="boy_belt">Boy Belt</option>
                        <option value="boy_clothe">Boy Clothe</option>
                        <option value="boy_footwear">Boy Footwear</option>
                        <option value="boy_headwear">Boy Headwear</option>
                        <option value="boy_jewellery">Boy Jewellery</option>
                        <option value="boy_tie">Boy Tie</option>
                        <option value="boy_underwear">Boy Underwear</option>
                        {/*girlItems */}
                        <option value="girl_bag">Girl Bag</option>
                        <option value="girl_belt">Girl Belt</option>
                        <option value="girl_clothe">Girl Clothe</option>
                        <option value="girl_footwear">Girl Footwear</option>
                        <option value="girl_headwear">Girl Headwear</option>
                        <option value="girl_jewellery">Girl Jewellery</option>
                        <option value="girl_tie">Girl Tie</option>
                        <option value="girl_underwear">Girl Underwear</option>
                        {/*menItems */}
                        <option value="men_bag">Men Bag</option>
                        <option value="men_belt">Men Belt</option>
                        <option value="men_clothe">Men Clothe</option>
                        <option value="men_footwear">Men Footwear</option>
                        <option value="men_headwear">Men Headwear</option>
                        <option value="men_jewellery">Men Jewellery</option>
                        <option value="men_tie">Men Tie</option>
                        <option value="men_underwear">Men Underwear</option>
                        {/*womenItems */}
                        <option value="women_bag">Women Bag</option>
                        <option value="women_belt">Women Belt</option>
                        <option value="women_clothe">Women Clothe</option>
                        <option value="women_footwear">Women Footwear</option>
                        <option value="women_headwear">Women Headwear</option>
                        <option value="women_jewellery">Women Jewellery</option>
                        <option value="women_tie">Women Tie</option>
                        <option value="women_underwear">Women Underwear</option>
                    </select>    
                    <input
                        className= "text-input"
                        type="text"
                        placeholder="Amount"                        
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                    />
                    <input
                        className= "text-input"
                        type="number"
                        placeholder="Stock"                        
                        value={this.state.stock}
                        onChange={this.onStockChange}
                    />
                    <input
                        className= "text-input"
                        type="text"
                        placeholder="Company Name"                        
                        value={this.state.company}
                        onChange={this.onCompanyChange}
                    />
                    <input
                        className= "text-input"
                        type="text"
                        placeholder="Product Information"                        
                        value={this.state.info}
                        onChange={this.onInfoChange}
                    />
                    <SingleDatePicker
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.focused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                    />
                   <div>
                        <input 
                            className= "text-input"
                            type="file"                            
                            onChange={this.onImageChange}
                        />
                        <div className="image-input">                            
                            <img src={this.state.imageUrl} height="200" width="200"/>                            
                        </div>                        
                    </div>
                <div>
                    {this.state.error && <p className="form__error">{this.state.error}</p>}
                </div>
                <div>
                    <button className="button">Save Product</button>                         
                </div>                   
            </form>                            
        );
    };
}

