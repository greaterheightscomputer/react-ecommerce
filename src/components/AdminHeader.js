import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const AdminHeader = ({ startLogout }) => (
    <header className="admin-header">        
        <div className="content-container">
            <div className="admin-header__container">
                <Link className="admin-header__title " to="/admin_dashboard">
                    <h1>E-Market</h1>
                </Link>
                <button className="button button--link" onClick={startLogout}>Logout</button>          
            </div>            
        </div>        
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(AdminHeader);