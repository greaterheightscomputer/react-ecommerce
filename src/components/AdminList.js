import React from 'react';
import { connect } from 'react-redux';
import AdminListItem from './AdminListItem';
import selectEcommerce from '../selectors/ecommerces';

const AdminList = (props) => (
    <div>
        <h1>Admin List</h1>
        {props.ecommerces.map((ecommerce) =>{
            return <AdminListItem key={ecommerce.id} {...ecommerce} />
        })}
        
    </div>
);

const mapStateToProps = (state) => {
    return {
        ecommerces: selectEcommerce(state.ecommerces, state.filters)        
    };
}

export default connect(mapStateToProps)(AdminList);