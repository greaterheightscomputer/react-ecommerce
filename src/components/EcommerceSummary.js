import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectEcommerce from '../selectors/ecommerces';
import selectEcommerceTotal from '../selectors/ecommerce-total'

export const EcommerceSummary = ({ ecommerceCount, ecommerceTotal }) => {
    const ecommerceWord = ecommerceCount === 1 ? 'product' : 'products';
    const formatedEcommerceTotal = 'â‚¦' + numeral(ecommerceTotal/100).format('0,0.00');
    return (
        <h1>Displaying {ecommerceCount} {ecommerceWord} totalling {formatedEcommerceTotal}</h1>
    );
};

const mapStateToProps = (state) => {
    const visibleEcommerces = selectEcommerce(state.ecommerces, state.filters);
    
    return {
        ecommerceCount: visibleEcommerces.length,
        ecommerceTotal: selectEcommerceTotal(visibleEcommerces)
    };
};

export default connect(mapStateToProps)(EcommerceSummary); 