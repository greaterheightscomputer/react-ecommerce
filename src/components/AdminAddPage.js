import React from 'react';
import { connect } from 'react-redux';
import AdminForm from './AdminForm';
import { startAddEcommerce } from '../actions/ecommerces';

export class AdminAddPage extends React.Component {
    onSubmit = (ecommerce) => {
        // console.log(ecommerce)
        // props.dispatch(addEcommerce(ecommerce));
        this.props.startAddEcommerce(ecommerce);
        this.props.history.push('/admin_dashboard');
    };
    render() {
        return (
            <div className="list-body">
                <div className="page-admin-header">
                    <div className="content-container">
                        <h1 className="page-admin-header__title"> Add Product</h1>
                    </div>
                </div>                
                <div className="content-container">
                    <AdminForm
                        onSubmit={this.onSubmit}
                    />
                </div>
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch) => ({
    startAddEcommerce: (ecommerce) => dispatch(startAddEcommerce(ecommerce))
});

export default connect(undefined, mapDispatchToProps)(AdminAddPage);