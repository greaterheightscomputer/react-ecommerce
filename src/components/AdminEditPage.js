import React from 'react';
import { connect } from 'react-redux';
import AdminForm from './AdminForm';
import { editEcommerce, removeEcommerce } from '../actions/ecommerces';

export const AdminEditPage = (props) => {
    // console.log(props)
    return(
    <div>
        <AdminForm 
            ecommerce={props.ecommerce}
            onSubmit={(ecommerce) => {
                props.dispatch(editEcommerce(props.ecommerce.id, ecommerce));
                props.history.push('/admin_dashboard');
                // console.log('updated', ecommerce);
            }}
        />
        <button onClick={() => {
            props.dispatch(removeEcommerce({ id: props.ecommerce.id }));            
            props.history.push('/admin_dashboard');
        }}>Remove</button>
    </div>
    );
};

const mapStateToProps = (state, props) => {
    return {
        ecommerce: state.ecommerces.find((ecommerce) => ecommerce.id === props.match.params.id)
    };
};

export default connect(mapStateToProps)(AdminEditPage);