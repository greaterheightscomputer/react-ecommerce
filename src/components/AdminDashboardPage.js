import React from 'react';
import AdminList from './AdminList';
import AdminListFilters from './AdminListFilters';
import EcommerceSummary from './EcommerceSummary';

const AdminDashboardPage = () => (
    <div>        
        <EcommerceSummary />
        <AdminListFilters />
        <AdminList />        
    </div>
);

export default AdminDashboardPage;