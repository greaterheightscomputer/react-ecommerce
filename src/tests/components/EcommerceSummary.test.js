import React from 'react';
import { shallow } from 'enzyme';
import { EcommerceSummary } from '../../components/EcommerceSummary';

test('should render EcommerceSummary with 1 product', () => {
    const wrapper = shallow(<EcommerceSummary ecommerceCount={1} ecommerceTotal={3450} />);
    expect(wrapper).toMatchSnapshot(); 
});

test('should render EcommerceSummary with multiple products', () => {
    const wrapper = shallow(<EcommerceSummary ecommerceCount={10} ecommerceTotal={78654} />);
    expect(wrapper).toMatchSnapshot();
});