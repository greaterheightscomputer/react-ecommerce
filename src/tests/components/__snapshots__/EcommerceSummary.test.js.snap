// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EcommerceSummary with 1 product 1`] = `
<h1>
  Displaying 
  1
   
  product
   totalling 
  ₦34.50
</h1>
`;

exports[`should render EcommerceSummary with multiple products 1`] = `
<h1>
  Displaying 
  10
   
  products
   totalling 
  ₦786.54
</h1>
`;
